### PR TITLE
Add support for orders custom data on transactions CSV

### PR DESCRIPTION
**Pull request by Betree**

### CODE REVIEW

File: src/server/controllers/account-transactions.js [hunk 1]
```diff
@@ -273,6 +274,7 @@ const csvMapping = {
   merchantId: (t) => get(t, 'merchantId'),
   orderMemo: (t) => get(t, 'order.memo'),
   orderProcessedDate: (t) => (t.order?.processedAt ? moment.utc(t.order.processedAt).format('YYYY-MM-DD') : ''),
+  orderCustomData: (t) => get(t, 'order.customData'),
```
Comment (znarf) [inline src/server/controllers/account-transactions.js:277]: Isn't custom data an object?

Comment (Betree) [reply src/server/controllers/account-transactions.js:277]: It does. In my tests `json2csv` properly stringified the value

Comment (Betree) [reply src/server/controllers/account-transactions.js:277]: It's from https://opencollective.slack.com/archives/C6JTTA4SK/p1696831552748589?thread_ts=1686331488.222289&cid=C6JTTA4SK, the data they're getting from this is a phone number collected via a custom field in the contribution flow.

The alternative would be to explode the `customData` object and have one column per field:
- Input: `{ phone: '0123456789', age: 43 }` 
- Output: 

| Custom data -> phone | customData -> age |
|--------|--------|
| 0123456789 | 43 |

Comment (znarf) [reply src/server/controllers/account-transactions.js:277]: Ok let's keep it as stringified JSON.

**Review by znarf — APPROVED** [review]

File: src/server/controllers/account-transactions.js (hunk 1 above)
Comment (Betree) [reply src/server/controllers/account-transactions.js:277]: If needed, we could implement a separate parameter to achieve that behavior of splitting into columns in the future

**Pull request merged**